### ISSUE TITLE
Problem with parent field of revisions

### ISSUE THREAD

**Issue by f-g-s-e** [directus/directus#8481]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

As described in the title, I have noticed an unexpected behavior in the valorization of the revision parent field under certain circumstances, better described in the section "To Reproduce".

### To Reproduce

- Create collection A
- Create collection B
- Create a many to one (M2O) relation from A to B
- Using the approach described [here](https://docs.directus.io/reference/api/introduction/#one-to-many-many-to-many) in section **Detailed**, do a PATCH request to API of collection B, trying to create an object of collection A. Here an example:

```
PATCH /items/collectionb/:idItemCollectionB

{
	"collectionA":{
			"create": [{"fieldCollecionA":"value"}]
		}
}
```

What happens is that in the revision of collection A the field parent is not set. Trying to perform the same action from the ui of the app everything works correctly, and the parent is set as expected.

### What version of Directus are you using?

v9.0.0-rc.95

### What version of Node.js are you using?

v16.8.0

### What database are you using?

mysql

### What browser are you using?

chrome

### What operating system are you using?

windows

### How are you deploying Directus?

locally

**Comment by paescuj** [comment]
Have you already checked how the PATCH request from the UI looks like (using web inspector)?

**Comment by azrikahar** [comment]
Hi, unfortunately seems like I was not able to reproduce this error. Would you mind checking out the following recording and let me know is there any difference in what I did compared to what you described? Just so we can avoid any misunderstanding from my end 👍

https://user-images.githubusercontent.com/42867097/135596196-b6716af5-5f54-49dc-beb2-02935128ddf6.mp4

and here's the payload returned (it was cut off in my recording):

![Insomnia_uNcjYtSK9Z](https://user-images.githubusercontent.com/42867097/135596272-03ecf3f9-9a85-4adb-93b2-e184f7d1f0aa.png)

**Comment by fincons-group** [comment]
> Hi, unfortunately seems like I was not able to reproduce this error. Would you mind checking out the following recording and letting me know is there any difference in what I did compared to what you described? Just so we can avoid any misunderstanding from my end 👍
> 
>  Tw4KPKp6Vb.mp4 
> and here's the payload returned (it was cut off in my recording):
> 
> ![Insomnia_uNcjYtSK9Z](https://user-images.githubusercontent.com/42867097/135596272-03ecf3f9-9a85-4adb-93b2-e184f7d1f0aa.png)

The steps in your video are ok. The problem is on DB, with the column **parent** of the **directus_revisions** table.  This column should contain the id of the revision of the operation on the collection A. Look at the picture below, this is what I have on DB table directus_revisions if I try from app:

![image](https://user-images.githubusercontent.com/91721986/135621461-c76caae3-b171-498a-95d7-bef707f0c7e5.png)

If I try to achive the same result from the api, using the approach [here](https://docs.directus.io/reference/api/introduction/#one-to-many-many-to-many) in the section **Detailed**, I get the result in the picture below:

![image](https://user-images.githubusercontent.com/91721986/135623419-cfd523bd-bba0-4d2d-a456-76fbfeee5713.png)

**Comment by paescuj** [comment]
> Have you already checked how the PATCH request from the UI looks like (using web inspector)?

**Comment by fincons-group** [comment]
> > Have you already checked how the PATCH request from the UI looks like (using web inspector)?

{
	"relation_with_a": [
		3,{
			"fielda":"item seven a"
		}
	]
}

**Comment by paescuj** [comment]
Thanks! Hmm, are there more requests or just this one?

**Comment by fincons-group** [comment]
> Thanks! Hmm, are there more requests or just this one?

Some GETs and a couple of PATCHes on this endpoint /users/me/track/page

**Comment by rijkvanzanten** [comment]
I think the difference here is in the specific O2M update syntax that's used:

The app uses the "inline array" version:

```
{
  "o2m_field": [1, 2, 3]
}
```

the OP example uses the "create/update/delete" syntax:

```json
{
  "o2m_field": {
    "create": [1, 2, 3]
  }
}
```

**Comment by azrikahar** [comment]
@rijkvanzanten That is true, I wasn't inspecting the DB so even though everything looks "fine" from the UI, but the revisions table does in fact _do not_ link the parent revision via API with the detailed create/update/delete syntax.

> The problem is on DB, with the column parent of the directus_revisions table. This column should contain the id of the revision of the operation on the collection A. 

Thanks for the clarification & screenshots! After looking into it, it is indeed a bug & will be fixed by #8553.
